File: app/src/main/java/es/etologic/mahjongscoring2/injection/AppComponent.java
package es.etologic.mahjongscoring2.injection;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
        AppActivitiesBuilder.class})
public interface AppComponent extends AndroidInjector<MahjongScoringApp> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<MahjongScoringApp> {}
}

File: app/src/main/java/es/etologic/mahjongscoring2/injection/AppComponent.kt
package es.etologic.mahjongscoring2.injection

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import es.etologic.mahjongscoring2.data.local_data_source.local.AppDataBaseModule
import javax.inject.Singleton

@Singleton
@Component(
    modules = [
        AndroidSupportInjectionModule::class,
        AppModule::class,
        AppActivitiesBuilder::class,
        AppDataBaseModule::class]
)
interface AppComponent : AndroidInjector<MahjongScoringApp> {
    
    @Component.Factory
    abstract class Factory : AndroidInjector.Factory<MahjongScoringApp>
}
